### DISCUSSION TITLE
Disable XOSC in favor of clock IC

### DISCUSSION THREAD

**Discussion by regicidalplutophage** [micropython/micropython#11174] [RP2040 / Pico] [answered]
Hi! I'm a newbie in both software and hardware design and I'm doing an RP2040 board.

To avoid messing up load capacitors in XOSC circuit, I'm thinking of using a CMOS clock IC.

RP2040 datasheet says:
> If the user already has an accurate clock source then it is possible to drive an external clock directly into XIN (aka XI),
> and disable the oscillator circuit. In this mode XIN can be driven at up to 50MHz.

But now I don't know how to achieve that. So, is it possible to build micropython with this setting?

**Comment by dhalbert** [comment]
> To avoid messing up load capacitors in XOSC circuit, I'm thinking of using a CMOS clock IC.

The RP2040 datasheet has a detailed discussion of how to pick the capacitors. Or you could use their reference design parts. If your oscillator is slow to start, you can use  `PICO_XOSC_STARTUP_DELAY_MULTIPLIER` to set a longer startup delay. I think this will be easier in the long run. Don't be afraid of the calculation -- it's straightforward.

**Reply by regicidalplutophage** [reply]
>Don't be afraid of the calculation -- it's straightforward.

Yeah, the rule of thumb calculation is indeed straightforward, but I'm not happy with my understanding of Pierce oscillator and "guessing" Cstray just doesn't feel satisfying as it's not something I'm going to fix later if I guess it wrong.

**Reply by dhalbert** [reply]
https://blog.adafruit.com/2012/01/24/choosing-the-right-crystal-and-caps-for-your-design/

**Comment by peterhinch** [comment]
Interesting and informative though these responses are, they don't actually answer the question asked by @regicidalplutophage.

As I read the OP he wants to use an oscillator chip rather than a crystal and wants to know how to disable the RP2 XO from MicroPython. Even if there is a register write that does this, I dout it could be issued quickly enough. Has anyone looked into this?

**Comment by peterhinch** [comment, marked as answer]
The RP2 datasheet states:
> Providing an external frequency source can be done in one of two ways: either by providing a clock source with a CMOS
output (3.3V square wave) into the XIN pin, or by using a 12MHz crystal...

This seems pretty clear. No need for special settings, and leave the `XOUT` pin open.

**Reply by regicidalplutophage** [reply]
I don't remember what exactly confused me me in the datasheet(maybe there was a register relating to xosc?), but this seems like a good answer
